src/stirling/bpf_tools/probe_specs/probe_specs.h

```diff
@@ -75,6 +75,9 @@ struct KProbeSpec {
   // kernels.
   bool is_optional = false;
 
+  // If the kernel function is not found, then this fallback function will be used instead.
+  struct KProbeSpec* fallback_probe = unique_ptr;
```

Comment:
Since this is still a pointer to a struct you will still leak memory. What I was imagining with my comment is:
```
std::unique_ptr<KProbeSpec> fallback_probe = nullptr;
```